platforms/native/platform-native/src/main/java/org/gradle/nativeplatform/internal/DefaultStaticLibraryBinarySpec.java

```diff
@@ -58,7 +59,10 @@ public void additionalLinkFiles(FileCollection files) {
 
     @Override
     public FileCollection getLinkFiles() {
-        return getFileCollectionFactory().create(new StaticLibraryLinkOutputs());
+        ConfigurableFileCollection result = getFileCollectionFactory().configurableFiles("Link files for " + getDisplayName());
+        result.from(getFileCollectionFactory().create(new StaticLibraryLinkOutputs()));
+        additionalLinkFiles.forEach(result::from);
```

Comment:
I think this might have a slightly different meaning now--`getLinkFiles()` will only contain the elements of `additionalLinkFiles` that were added before `getLinkFiles` is called. 

It's hard to tell if this actually makes a difference because there are so many places we call `getLinkFiles` in a deferred way. 

Would something like this also work for CC?
```
result.from(additionalLinkFiles)
```